src/components/pdf-viewer/PdfViewer.tsx

```diff
@@ -0,0 +1,46 @@
+import 'react-pdf/dist/esm/Page/TextLayer.css';
+import 'react-pdf/dist/esm/Page/AnnotationLayer.css';
+
+import { Document, Page, pdfjs } from 'react-pdf';
+
+import styles from './PdfViewer.module.less';
+import { useState } from 'react';
+
+pdfjs.GlobalWorkerOptions.workerSrc = 'pdf.worker.min.js';
+
+interface Props {
+  documentUrl: string,
+  width: number,
+  height: number,
+  testId: string
+}
+
+const PdfViewer = (props: Props) => {
+  const [numPages, setNumPages] = useState(1);
+
+  function onDocumentLoadSuccess({ numPages }) {
+    setNumPages(numPages);
+  }
```

Comment:
my suggestion:

```
const [pdfPages, setPdfPages] = useState<Page>([]);

const onDocumentLoadSuccess = (numPages: number) => {
    setPdfPages(new Array(numPages).map(_, index) => <Page ... />);
}
```

it would simplify the return

Reply:
Done, it wasn't that easy so I leave this issue unresolved.

Reply:
fine, I've left one more comment about it below